### PR TITLE
Add: no-floating-promises linting rule

### PR DESCRIPTION
**Pull request by Leo310**
This rule reports when a Promise is created and not properly handled.  For example, when you forget using await.

### CODE REVIEW

**Comment by dhenkel92** [comment]
👍

**Comment by Jonasdoubleyou** [comment]
Maybe we could also try https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-misused-promises.md

**Comment by Leo310** [comment]
@Jonasdoubleyou @dhenkel92 putting void everywhere means more boilerplate code but it overall seems cleaner.  Ignoring a specific function with eslint isn't really possible when they are imported from libraries. 
What do you think?

File: common/interest-confirmation/tutoring/persistence/change-status.ts [hunk 1]
```diff
@@ -26,5 +30,5 @@ export async function changeStatus(token: string, status: InterestConfirmationSt
 
     //transaction log
     const transactionLog = getTransactionLog();
-    transactionLog.log(new PupilInterestConfirmationRequestStatusChangeEvent(confirmationRequest, previousStatus));
-}
\ No newline at end of file
+    void transactionLog.log(new PupilInterestConfirmationRequestStatusChangeEvent(confirmationRequest, previousStatus));
```
Comment (Jonasdoubleyou) [inline common/interest-confirmation/tutoring/persistence/change-status.ts:33]: await ;)

File: graphql/authentication.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ export async function loginAsUser(user: User, context: GraphQLContext, noSession
     }
 
     if (!noSession) {
-        userSessions.set(context.sessionToken, context.user);
+        void userSessions.set(context.sessionToken, context.user);
```
Comment (Jonasdoubleyou) [inline graphql/authentication.ts:128]: await :)

File: graphql/pupil/mutations.ts [hunk 3]
```diff
@@ -181,7 +181,7 @@ export async function updatePupil(
     });
 
     if (pupil.registrationSource !== 'plus' && registrationSource === 'plus') {
-        Notification.actionTaken(pupil, 'pupil_joined_plus', {});
+        void Notification.actionTaken(pupil, 'pupil_joined_plus', {});
```
Comment (Jonasdoubleyou) [inline graphql/pupil/mutations.ts:184]: await :)

File: graphql/secret/mutation.ts [hunk 4]
```diff
@@ -43,7 +43,7 @@ export class MutateSecretResolver {
     @Authorized(Role.USER)
     async meChangeEmail(@Ctx() context: GraphQLContext, @Arg('email') email: string) {
         const user = await getSessionUser(context);
-        requestToken(user, 'user-email-change', '/start', email);
+        void requestToken(user, 'user-email-change', '/start', email);
```
Comment (Jonasdoubleyou) [inline graphql/secret/mutation.ts:46]: await :)

File: graphql/tutoring_interest_confirmation/mutations.ts [hunk 5]
```diff
@@ -34,7 +34,7 @@ async function changeStatus(token: string, status: InterestConfirmationStatus) {
     });
 
     const pupil = await getPupil(confirmation.pupilId);
-    logTransaction('pupilInterestConfirmationRequestStatusChange', pupil, {
+    void logTransaction('pupilInterestConfirmationRequestStatusChange', pupil, {
```
Comment (Jonasdoubleyou) [inline graphql/tutoring_interest_confirmation/mutations.ts:37]: await :)

File: graphql/tutoring_interest_confirmation/mutations.ts [hunk 6]
```diff
@@ -73,7 +73,7 @@ export class MutateTutoringInterestConfirmationResolver {
             data: { token: generateToken(), pupilId, status, invalidated: false },
         });
 
-        logTransaction('pupilInterestConfirmationRequestStatusChange', pupil, {
+        void logTransaction('pupilInterestConfirmationRequestStatusChange', pupil, {
```
Comment (Jonasdoubleyou) [inline graphql/tutoring_interest_confirmation/mutations.ts:76]: await :)

File: jobs/manualExecution.ts [hunk 7]
```diff
@@ -19,55 +19,55 @@ import { runInterestConfirmations } from '../common/match/pool';
 export const executeJob = async (job) => {
     switch (job) {
         case 'screeningReminderJob': {
-            screeningReminderJob(getManager());
+            void screeningReminderJob(getManager());
```
Comment (Jonasdoubleyou) [inline jobs/manualExecution.ts:22]: await

File: jobs/periodic/course-reminder/index.ts [hunk 8]
```diff
@@ -57,15 +56,15 @@ async function sendUpcomingCourseReminders(manager: EntityManager) {
...
         if (moment(firstLecture.start).isSame(dayAfterTomorrow, 'day')) {
             // first lecture is in two days
-            logger.info("Found lecture on (sub)course " + course.name + " with start date in 2 days: " + firstLecture.start.toDateString());
-            logger.info("Sending reminders to instructor and " + subcourse.subcourse_participants_pupil.length + " participants");
+            logger.info('Found lecture on (sub)course ' + course.name + ' with start date in 2 days: ' + firstLecture.start.toDateString());
+            logger.info('Sending reminders to instructor and ' + subcourse.subcourse_participants_pupil.length + ' participants');
 
             // notify instructor
-            sendCourseUpcomingReminderInstructor(firstLecture.student, course, subcourse, firstLecture.start);
+            void sendCourseUpcomingReminderInstructor(firstLecture.student, course, subcourse, firstLecture.start);
```
Comment (Jonasdoubleyou) [inline jobs/periodic/course-reminder/index.ts:63]: await

File: jobs/periodic/course-reminder/index.ts [hunk 9]
```diff
@@ -57,15 +56,15 @@ async function sendUpcomingCourseReminders(manager: EntityManager) {
...
 
             // notify all participants
             for (let i = 0; i < subcourse.subcourse_participants_pupil.length; i++) {
-                sendCourseUpcomingReminderParticipant(subcourse.subcourse_participants_pupil[i].pupil, course, subcourse, firstLecture.start);
+                void sendCourseUpcomingReminderParticipant(subcourse.subcourse_participants_pupil[i].pupil, course, subcourse, firstLecture.start);
```
Comment (Jonasdoubleyou) [inline jobs/periodic/course-reminder/index.ts:67]: await

File: jobs/scheduler.ts [hunk 10]
```diff
@@ -55,35 +53,33 @@ function executeJob(job: (manager: EntityManager) => Promise<void>, jobConnectio
...
 }
 
 export async function unscheduleAllJobs() {
-    scheduledJobs.forEach(j => j.stop());
+    scheduledJobs.forEach((j) => j.stop());
 }
 
 export async function shutdownConnection() {
-    jobConnection?.close();
+    void jobConnection?.close();
```
Comment (Jonasdoubleyou) [inline jobs/scheduler.ts:84]: await

File: web/index.ts [hunk 11]
```diff
@@ -88,7 +88,7 @@ createConnection()
         configureRegistrationAPI();
         configureMentoringAPI();
         configureExpertAPI();
-        configureApolloServer();
+        void configureApolloServer();
```
Comment (Jonasdoubleyou) [inline web/index.ts:91]: await

**Review by Jonasdoubleyou — COMMENTED** [review]
Actually most of them aren't false positives 😅

File: common/interest-confirmation/tutoring/persistence/change-status.ts (hunk 1 above)
Comment (Leo310) [reply common/interest-confirmation/tutoring/persistence/change-status.ts:33]: aber ist das da nicht völlig egal? ist doch nur logging oder?

File: jobs/manualExecution.ts (hunk 7 above)
Comment (Leo310) [reply jobs/manualExecution.ts:22]: wenn hier, warum dann bei den anderen cases nicht?

Comment (Jonasdoubleyou) [reply jobs/manualExecution.ts]: da auch, war zu faul hier an jede Zeile zu kommentieren ;)

Comment (Leo310) [reply jobs/manualExecution.ts]: Verstehe 🦥

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**